### PR TITLE
Feature/zen 25843

### PR DESCRIPTION
**Pull request by lundybernard**
Includes Flake8 linter cleanup

adds checking for original sender of snmp traps, and sets the device ID to the sources IP address if found.

### CODE REVIEW

File: Products/ZenEvents/zentrap.py [hunk 1]
```diff
@@ -39,26 +39,32 @@
...
 from Products.ZenUtils.captureReplay import CaptureReplay
 from Products.ZenEvents.EventServer import Stats
-from Products.ZenUtils.Utils import unused
 from Products.ZenEvents.TrapFilter import TrapFilter, TrapFilterError
 from Products.ZenEvents.ZenEventClasses import Clear, Critical, Error
 from Products.ZenUtils.Utils import unused, zenPath
 from Products.ZenCollector.services.config import DeviceProxy
 from Products.ZenHub.services.SnmpTrapConfig import User
-unused(Globals, DeviceProxy, User)
+unused(Globals, DeviceProxy, User, Error, zenPath)
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: Generally, `unused` is for imports that are necessary for their side-effects.  I know `zenPath` has no side-effects and so its import can be removed.  You'll have to test whether `Error` is the same.

File: Products/ZenEvents/zentrap.py [hunk 2]
```diff
@@ -383,20 +412,32 @@ def oid2name(self, oid, exactMatch=True, strip=False):
...
-            ipv6_socket_address = c.cast(transport_data, c.POINTER(sockaddr_in6)).contents
+            ipv6_socket_address = c.cast(transport_data,
+                                         c.POINTER(sockaddr_in6)
+                                         ).contents
             if ipv6_socket_address.family == socket.AF_INET6:
-                self.log.debug("pre_parse: IPv6 %s" % (socket.inet_ntop(socket.AF_INET6, ipv6_socket_address.addr)))
+                self.log.debug("pre_parse: IPv6 %s",
+                               socket.inet_ntop(socket.AF_INET6,
+                                                ipv6_socket_address.addr)
+                               )
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: This is an example of why I don't like the align with the opening '(' way of formatting.  I think the following is far more readable and is still pep8 compliant:
```
    self.log.debug(
        "pre_parse: IPV6 %s",
        socket.inet_ntop(socket.AF_INET6, ipv6_socket_address.addr)
    )
```
or, if it's still too wide:
```
    self.log.debug(
        "pre_parse: IPV6 %s",
        socket.inet_ntop(
            socket.AF_INET6, ipv6_socket_address.addr
        )
    )
```

File: Products/ZenEvents/zentrap.py [hunk 3]
```diff
@@ -590,11 +643,18 @@ def _add_varbind_detail(self, result, oid, value):
...
+        result["device"] = addr[0]
 
         variables = self.getResult(pdu)
 
-        result["device"] = addr[0]
+        self.log.debug("SNMPv1 pdu has agent_addr: %s",
+                       str(hasattr(pdu, 'agent_addr')))
+
+        if hasattr(pdu, 'agent_addr'):
+            origin = '.'.join([str(i) for i in pdu.agent_addr])
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: List comprehension is unnecessary here.  A generator comprehension will work just fine.  E.g.
```
    origin = '.'.join(str(i) for i in pdu.agent_addr)
```

Comment (lundybernard) [inline Products/ZenEvents/zentrap.py]: fixed

File: Products/ZenEvents/zentrap.py [hunk 4]
```diff
@@ -60,15 +60,14 @@
...
 from Products.ZenEvents.EventServer import Stats
 from Products.ZenEvents.TrapFilter import TrapFilter, TrapFilterError
-from Products.ZenEvents.ZenEventClasses import Clear, Critical, Error
-from Products.ZenUtils.Utils import unused, zenPath
+from Products.ZenEvents.ZenEventClasses import Clear, Critical
+from Products.ZenUtils.Utils import unused
 from Products.ZenCollector.services.config import DeviceProxy
 from Products.ZenHub.services.SnmpTrapConfig import User
-unused(Globals, DeviceProxy, User, Error, zenPath)
+unused(Globals, DeviceProxy, User)  # , Error)
```
Comment (jpeacock-zenoss) [inline Products/ZenEvents/zentrap.py]: Commented out code should be removed.

Comment (lundybernard) [inline Products/ZenEvents/zentrap.py]: removed in 23a1a29

**Comment by lundybernard** [comment]
retest this

**Comment by lundybernard** [comment]
retest this

**Pull request merged**